File: backend-node/tests/test-registration.js
// Test registration with new user
const http = require('http');

function makeRequest(options, data = null) {
  return new Promise((resolve, reject) => {
    const req = http.request(options, (res) => {
      let body = '';
      res.on('data', (chunk) => body += chunk);
      res.on('end', () => {
        try {
          const parsed = JSON.parse(body);
          resolve({ status: res.statusCode, data: parsed });
        } catch (e) {
          resolve({ status: res.statusCode, data: body });
        }
      });
    });
    
    req.on('error', reject);
    
    if (data) {
      req.write(JSON.stringify(data));
    }
    req.end();
  });
}

async function testNewRegistration() {
  console.log('🧪 Testing Registration with New User\n');
  
  const registerOptions = {
    hostname: 'localhost',
    port: 5000,
    path: '/api/auth/register',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    }
  };
  
  const registerData = {
    name: 'Test User New',
    email: `test${Date.now()}@example.com`,
    password: 'SecurePass123'
  };
  
  console.log('Registering user:', registerData);
  
  try {
    const result = await makeRequest(registerOptions, registerData);
    console.log(`\nStatus: ${result.status}`);
    console.log(`Response:`, result.data);
  } catch (error) {
    console.error('❌ Test failed:', error.message);
  }
  
  process.exit(0);
}

testNewRegistration();

File: test_registration.py
#!/usr/bin/env python3
"""
Quick test script to verify registration API endpoint
"""

import requests
import json

def test_registration():
    """Test the registration endpoint directly"""
    
    # Test data
    test_user = {
        "username": "testuser123",
        "email": "testuser123@example.com", 
        "password": "TestPass123!"
    }
    
    print("🧪 Testing Registration API Endpoint")
    print(f"📤 Sending POST to http://localhost:5000/api/auth/register")
    print(f"📦 Payload: {json.dumps(test_user, indent=2)}")
    
    try:
        # Make the request
        response = requests.post(
            "http://localhost:5000/api/auth/register",
            json=test_user,
            headers={"Content-Type": "application/json"},
            timeout=10
        )
        
        print(f"\n📡 Response Status: {response.status_code}")
        print(f"📡 Response Headers: {dict(response.headers)}")
        
        try:
            response_data = response.json()
            print(f"📦 Response Data: {json.dumps(response_data, indent=2)}")
        except:
            print(f"📦 Response Text: {response.text}")
        
        if response.status_code == 201:
            print("✅ Registration successful!")
            return True
        else:
            print(f"❌ Registration failed with status {response.status_code}")
            return False
            
    except requests.exceptions.ConnectionError:
        print("❌ Connection error - is the backend server running on port 5000?")
        return False
    except requests.exceptions.Timeout:
        print("❌ Request timeout")
        return False
    except Exception as e:
        print(f"❌ Unexpected error: {e}")
        return False

if __name__ == "__main__":
    test_registration()
